daemon/src/peer.rs

```diff
@@ -230,15 +149,29 @@ impl P2PActor {
                 panic!("For security reasons, please make sure to set the key file to user-readable only (set the permissions to 600).");
             }
             info!("Re-using existing keypair");
-            let bytes = std::fs::read(keyfile).expect("Failed to read key file");
-            Keypair::from_protobuf_encoding(&bytes).expect("Failed to deserialize key file")
+            let mut bytes = [0; 32];
+            let mut file = File::open(keyfile).expect("Failed to open key file");
+            file.read_exact(&mut bytes)
+                .expect("Failed to read key file");
+
+            let mut passphrase = vec![];
+            file.read_to_end(&mut passphrase)
+                .expect("Failed to read from key file");
+
+            (
+                SecretKey::from_bytes(&bytes),
+                String::from_utf8(passphrase).expect("Passphrase should be valid UTF-8"),
+            )
         } else {
             info!("Generating new keypair");
-            // TODO: Is this the best algorithm?
-            let keypair = Keypair::generate_ed25519();
-            let bytes = keypair
-                .to_protobuf_encoding()
-                .expect("Failed to serialize keypair");
+            let secret_key = SecretKey::generate(rand::rngs::OsRng);
+
+            let bytes = secret_key.to_bytes();
+
+            let mut rng = rand::rngs::OsRng;
+            let passphrase: String = (0..64)
+                .map(|_| rng.sample(rand::distributions::Alphanumeric) as char)
+                .collect();
 
             let mut file = OpenOptions::new()
                 .create(true)
```

Comment:
we noticed that the truncate of the existing code might not be necessary, because there's an existence check anyways